src/policyengine/report_elements/single_simulation.py

```
@@ -0,0 +1,106 @@
+from __future__ import annotations
+
+from typing import Any, Iterable
+
+import numpy as np
+import pandas as pd
+
+from microdf import MicroDataFrame
+
+from policyengine.models.aggregate import Aggregate, AggregateMetric
+from policyengine.models.reports import ReportElement
+from policyengine.models.simulation import Simulation
+from policyengine.models.single_year_dataset import SingleYearDataset
+
+
+class AggregateReportElement(ReportElement):
+    """Compute aggregates for a variable, optionally grouped by a filter variable.
+
+    Returns a list of `Aggregate` records. Call `Aggregate.to_dataframe(records)` to
+    obtain a DataFrame for charting.
+    """
+
+    simulation: Simulation
+    variable: str
+    metric: AggregateMetric = AggregateMetric.MEAN
+    entity_level: str = "person"
+    filter_variable: str | None = None
+
+    def run(self) -> list[Aggregate]:
+        sim = self.simulation
+        if not isinstance(sim.result, (dict, SingleYearDataset)) and (
+            getattr(sim, "result", None) is None
+        ):
+            return []
+
+        data: SingleYearDataset | None = None
+        if isinstance(sim.result, dict):  # not expected but defensive
+            return []
+        else:
+            # Simulation.result is a Dataset with .data being SingleYearDataset
+            try:
+                data = sim.result.data  # type: ignore[attr-defined]
+            except Exception:
+                # If result is None or not shaped as expected
+                return []
+        if not isinstance(data, SingleYearDataset):
+            return []
```

__question:__ I'm surprised these fail silently